mmv1/products/networkservices/product.yaml

```diff
@@ -16,7 +16,7 @@ name: 'NetworkServices'
 display_name: 'Network services'
 versions:
   - name: 'beta'
-    base_url: 'https://networkservices.googleapis.com/v1/'
+    base_url: 'https://networkservices.googleapis.com/v1beta1/'
```

Comment:
This change is what's causing the new test failures. I think (based on looking at your API service config) that the wasm endpoints are being served on `/v1/` so you should be able to revert.

Reply:
/v1/ may not be available until Q1 2025. We need to stay with `v1beta1`.

Reply:
The [discovery doc](https://networkservices.googleapis.com/$discovery/rest?version=v1) says that the wasm plugin endpoints are available under `/v1/` so it might be worth trying. If it works, it'll save you from having to fix all the tests that are broken when connecting to `/v1beta1/`

Reply:
Running the v1 endpoints for wasmPlugins causes a 403 error: "WasmPlugins v1 api is not yet available. Please use the v1beta1 API." so I guess there is no way around that.

Most of the broken tests are related to NetworkServicesEdgeCache resources, and the docs lists only [v1](https://cloud.google.com/media-cdn/docs/reference/rest/v1/projects.locations.edgeCacheOrigins) and [v1alpha1](https://cloud.google.com/media-cdn/docs/reference/rest/v1alpha1/projects.locations.edgeCacheOrigins) endpoints for these, so maybe we could use v1alpha1 endpoints for the beta provider?

Reply:
It's OK to use a higher-versioned endpoint (like using GA for beta as has been used here) but going the other way is not. This is the first time I've encountered a split where one resource is available in GA but not beta, while another resource is available in beta but not GA. I'll talk to the team and see if there's a precedent for how to handle the situation.

Reply:
> Most of the broken tests are related to `NetworkServicesEdgeCache*` resources, and the docs lists only [`v1`](https://cloud.google.com/media-cdn/docs/reference/rest/v1/projects.locations.edgeCacheOrigins) and [`v1alpha1`](https://cloud.google.com/media-cdn/docs/reference/rest/v1alpha1/projects.locations.edgeCacheOrigins) endpoints for these, so maybe we could use `v1alpha1` endpoints for the beta provider?
I can't image how NetworkServicesEdgeCache tests may be impacted, as it would require WasmAction resource (a connection between WasmPlugin and EdgeCacheService) which stays in v1alpha and is not a part of the current task (you have access to WasmAction API because your test projects are allowlisted). 

At this moment WasmPlugin is intented to be used with LbTrafficExtension and LbRouteExtensions only.

Reply:
> I can't image how NetworkServicesEdgeCache tests may be impacted

Those tests are unchanged, but Magic Modules only supports a single base URL per version (beta or GA) and this PR changes the base URL for the beta version from `/v1/` to `/v1beta/`.

@matheusaleixo-cit after discussing this with the rest of the team the solution is to move this new resource to a new product (say, `networkservicesplugins`). That should be a quick change:

- Create the new `mmv1/products/networkservicesplugins` directory
- Move `WasmPlugin.yaml` to it
- Copy  `mmv1/products/networkservices/product.yaml` to it, with the beta base url set to `/v1beta`
- Revert the change to `mmv1/products/networkservices/product.yaml`